### DISCUSSION TITLE
elementary OS: under the hood

### DISCUSSION THREAD

**Discussion by alcinnz** [elementary/edw#42] [Accepted Talks]
Speaker: @alcinnz

This talk would describe how elementary OS works focusing on how input becomes output. It will be fairly highlevel and technical focusing on how all the different components fit together.

The primary intent would be to provide a useful perspective from which to understand the APIs used by app developers.

**Comment by danirabbit** [comment]
@alcinnz Do you think you could elaborate a bit more on how you think this information helps developers write apps? What problems does it help them understand and solve, etc? Is this something like how/when to use the various event signals for Gtk.Widget vs GLib.Action or can you be a bit more specific about what you're looking to cover?

**Reply by alcinnz** [reply]
Fair enough.

Mostly it'll serve as a tour of all the main libraries app developers use, and what they're useful for. I'll certainly explain the GLib mainloop, why some APIs are in the Gdk as opposed to Gtk namespace, and what the difference between GTK3 & GTK4 is.

Here's the order in which I'm planning to mention the libraries/tools:

1. C (for sake of contrast)
2. GObject
3. GLib
4. Vala
5. Window management & Linux drivers
6. GDK
7. GTK
8. GTK CSS
9. graphics (I go back & forth on how much detail to put here, I'm currently thinking "not much")
10. Window management & Linux drivers
11. GIO

**Comment by cassidyjames** [comment]
@alcinnz sorry for the delay in getting back to you, but would you still be able to put a talk together on this topic in the next week or two for edw? If the scope is too ambitious, a reduction would be fine, too.

**Reply by alcinnz** [reply]
Sure! How long should the talk be?

**Reply by cassidyjames** [reply]
@alcinnz it's flexible, but we're aiming for under 30 mins.
